baselines/flwr_baselines/publications/fedavg_mnist_2nn/README.md

```diff
@@ -0,0 +1,86 @@
+
+**Paper Abstract:** 
+
+<center>
+<i>Modern mobile devices have access to a wealth
+of data suitable for learning models, which in turn
+can greatly improve the user experience on the
+device. For example, language models can improve speech recognition and text entry, and image models can automatically select good photos.
+However, this rich data is often privacy sensitive,
+large in quantity, or both, which may preclude
+logging to the data center and training there using
+conventional approaches. We advocate an alternative that leaves the training data distributed on
+the mobile devices, and learns a shared model by
+aggregating locally-computed updates. We term
+this decentralized approach Federated Learning.
+We present a practical method for the federated
+learning of deep networks based on iterative
+model averaging, and conduct an extensive empirical evaluation, considering five different model architectures and four datasets. These experiments
+demonstrate the approach is robust to the unbalanced and non-IID data distributions that are a
+defining characteristic of this setting. Communication costs are the principal constraint, and
+we show a reduction in required communication
+rounds by 10–100× as compared to synchronized
+stochastic gradient descent</i>
+</center>
+
+**Paper Authors:** 
+
+H. Brendan McMahan, Eider Moore, Daniel Ramage, Seth Hampson, and Blaise Aguera y Arcas.
+
+
+Note: If you use this implementation in your work, please remember to cite the original authors of the paper. 
+
+**[Link to paper.](https://arxiv.org/abs/1602.05629)**
+
+## Training Setup
+
+### 2NN Architecture
+
+The 2NN architecture is detailed in the paper and used to create the **Federated Averaging MNIST** baseline.
+
+| Layer | Details|
+| ----- | ------ |
+| 1 | FC(784, 200) <br/> ReLU |
+| 2 | FC(200, 200) <br/> ReLU  |
+| 3 | FC(200, 10) |
+
+### Training Paramaters
+
+| Description | Value |
+| ----------- | ----- |
+| loss | cross entropy loss |
+| optimizer | SGD |
+| learning rate | 0.1 (by default) |
+| local epochs | 5 (by default) |
+| local batch size | 10 (by default) |
+
+## Running experiments
+
+The `config.yaml` file containing all the tunable hyperparameters and the necessary variables can be found under the `conf` folder.
+[Hydra](https://hydra.cc/docs/tutorials/) is used to manage the different parameters experiments can be ran with. 
+
+To run using the default parameters, just enter `python main.py`, if some parameters need to be overwritten, you can do it like in the following example: 
+
+```sh
+python main.py num_epochs=5 num_rounds=1000 iid=True
+``` 
+
+Results will be stored as timestamped folders inside either `outputs` or `multiruns`, depending on whether you perform single- or multi-runs. 
+
+### Example output
+
+To help visualize results, the script also plots evaluation curves. Here is an example:
+
+<p align="center">
+      <img src="docs/centralized_metrics.png" alt="Centralized evaluation results" width="400">
```

Comment:
@apzl I think you forgot to commit that file with the results, I will have a look at the rest of the PR but it looks good! Sorry for the time it took us to review

Reply:
How to plot the results? Should I write a script for that myself, or the same script in FedAvg MNIST would work?

Reply:
We have changed the structure of the baselines project since you've opened your PR, would it be ok if we merge this PR into a new branch in order to work on it ourselves and make the necessary arrangements? Or do you prefer doing the arrangements yourself?

Reply:
Yeah. You can make changes to accomodate the latest structure.